src/components/uploadPicture/uploadImage.tsx

```diff
@@ -0,0 +1,190 @@
+    setShowStart(false);
+    setShowTakePicture(true);
+  };
+
+  const handleFile = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const file = event.target.files?.[0];
+    if (file) {
+      const reader = new FileReader();
+      reader.onload = () => {
+        const fileContent = reader.result;
+        if (fileContent instanceof ArrayBuffer) {
+          const fileBlob = new Blob([fileContent]);
+          setCapturedImage(fileBlob);
+        }
+      };
+      reader.readAsArrayBuffer(file);
+
+      setShowTakePicture(false);
+      setShowUploadPicture(true);
+    }
+  };
+
+  const onCancelTakePicture = () => {
+    setShowStart(true);
+    setShowTakePicture(false);
+  };
+
+  const onCancelCaptureImage = () => {
+    setShowTakePicture(true);
+    setShowUploadPicture(false);
+  };
+
+  const onClickUploadPicture = () => {
+    setShowUploadPicture(false);
+    setShowStart(true);
+    setStartButtonDisabled(true);
+    setUploading(true);
+  };
+
+  const getPresignedUrlForUpload =
+    trpc.file.getPresignedUrlForFileUpload.useMutation();
+  const finishFileUpload = trpc.file.finishFileUpload.useMutation();
+
+  const uploadImageToCloud = useCallback(async () => {
+    if (!capturedImage) return;
+
+    const { key, url } = await getPresignedUrlForUpload.mutateAsync({
+      type: capturedImage.type || "application/octet-stream",
+    });
+
+    await axios.put(url, capturedImage);
+
+    await finishFileUpload.mutateAsync({
+      key,
+      filename: fileName,
+    });
+
+    setUploadSuccess(true);
+
+    onUpload(key);
+
+    setShowStart(true);
+    setStartButtonDisabled(false);
+    setUploading(false);
+  }, [
+    capturedImage,
+    fileName,
+    getPresignedUrlForUpload,
+    finishFileUpload,
+    onUpload,
+  ]);
+
+  useEffect(() => {
+    if (uploading) {
+      void uploadImageToCloud();
+    }
+  }, [uploading, uploadImageToCloud]);
+
+  useEffect(() => {
+    if (uploadSuccess) {
+      const timeout = setTimeout(() => {
+        setUploadSuccess(false);
+      }, 2000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [uploadSuccess]);
+
+  return (
+    <div>
+      {showStart && (
+        <button
+          className={
+            startButtonDisabled
+              ? styles.startButtonDisabled
+              : styles.startButton
+          }
+          onClick={onClickStartButton}
+          disabled={startButtonDisabled}
+        >
+          <div className={styles.startButtonContent}>
+            <CameraIcon />
+            <span>{title}</span>
+          </div>
+        </button>
+      )}
+      {showTakePicture && (
+        <>
+          <input
+            type="file"
+            accept="image/*;capture=camera"
+            capture="environment"
+            onChange={handleFile}
+            className={styles.startButton}
+          />
+          <button className={styles.startButton} onClick={onCancelTakePicture}>
+            Cancel
+          </button>
+        </>
+      )}
```

Comment:
it looks like this requires two clicks to open the camera, could we get rid of the first step?

Reply:
it looks like this still requires two clicks?

Reply:
At my end, it does not require two clicks so I am not sure what might be the issue.

Reply:
sorry, maybe this recording makes it more clear:


https://github.com/sfbrigade/compass/assets/7410405/d92425c2-4821-483c-9e73-951c183ef584

it looks like it requires two clicks until the file browser pops up?

that being said it's not the biggest deal and we can fix it later, if you [fix the lock file](https://github.com/sfbrigade/compass/pull/139#discussion_r1267503408) I think this would be good to merge

Reply:
@codetheweb Oh i see. Got it. That is because I hid the file input button until the user has clicked the button. I have tried some other ways of triggering click on the file input dialog but it did not work. Maybe, ticket this additional functionality for later.

Reply:
@jonahchoi could you help with this? I think this PR has taken too long lol

Reply:
I'll go ahead and fix the lock file, merge this in, then create some new issues 👍 

thanks for the work!